### DISCUSSION TITLE
Entity hierarchy?

### DISCUSSION THREAD

**Discussion by hw762** [Quillraven/Fleks#95] [General]
I am trying to find out a way to express one-to-many relationship in Fleks. Say I have multiple `Units`, each can have a number of `Weapons`. Each `Unit` and `Weapon` also have a number of components.

Is this possible? I don't see a way to, say, add a "child entity".

**Comment by Quillraven** [comment]
Not sure if this is a Fleks specific question. The way I usually solve entity relations is via components. I don't know if this is the best way to do it but it works for me.

As an idea how I would solve your problem:
1) Units are entities with components
2) Weapons are entities with components
3) Create an `Equipment` component which has a list of entities (=weapons)
4) Add this component to a unit and you have your connection between unit and weapons

I use a similar approach for collision handling. I usually have a Collision component that is added to an entity and contains a list of entities that it collided with within one frame.
You can have a look at this approach in my MysticGarden game: https://github.com/Quillraven/MysticGarden/blob/kotlin/core/src/main/kotlin/com/github/quillraven/mysticgarden/component/Collision.kt

----

Anyway, I suggest to solve entity relations via components as well. That should be the most flexible solution.

**Reply by hw762** [reply]
Shoulc I be using `EntityBag`? I can't seem to construct it. Or is it fine to just use `List` instead?

**Reply by Quillraven** [reply]
`EntityBag` is just an interface. You can use `MutableEntityBag` which is the implementation or just go with `List`. I think there is no big difference ;)
